src/WpfMath/TexFormulaParser.cs

```diff
@@ -609,7 +609,15 @@ private Color ReadColorModelData(SourceSpan value, ref int position)
                 // Predefined formula was found.
                 var predefinedFormula = factory(formulaSource);
                 var atom = AttachScripts(formula, value, ref position, predefinedFormula.RootAtom, true, environment);
-                formula.Add(atom, formulaSource);
+                if (atom.Source != null)
+                {
+                    var source = new SourceSpan(formulaSource.Source, formulaSource.Start, predefinedFormula.RootAtom.Source.Length + 1);
```

Comment:
Would `commandSpan` work here? If not, then please leave a comment about what're you trying to achieve here.